### PR TITLE
Keyboard: Planck THK

### PR DESCRIPTION
**Pull request by Erovia**
## Description

It's alive!

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/planck/thk/keymaps/thk/keymap.c [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+      if (record->event.pressed) {
+        layer_off(_RAISE);
+        layer_off(_LOWER);
+        layer_off(_ADJUST);
+        layer_on(_PLOVER);
+        if (!eeconfig_is_enabled()) {
+          eeconfig_init();
+        }
+        keymap_config.raw = eeconfig_read_keymap();
+        keymap_config.nkro = 1;
```
Comment (drashna) [inline keyboards/planck/thk/keymaps/thk/keymap.c:204]: Doesn't support NKRO, might as well remove the steno/plover stuff completely.

Comment (Erovia) [reply keyboards/planck/thk/keymaps/thk/keymap.c]: Good point, removed them.

File: keyboards/planck/thk/rules.mk [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# MCU name
+MCU = atmega32a
+#
```
Comment (fauxpark) [inline keyboards/planck/thk/rules.mk:18]:
```suggestion

```

File: keyboards/planck/thk/info.json [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        { "label": "k38", "matrix": [3, 8], "w": 1, "x": 8, "y": 3 },
+        { "label": "k39", "matrix": [3, 9], "w": 1, "x": 9, "y": 3 },
+        { "label": "k3a", "matrix": [3, 10], "w": 1, "x": 10, "y": 3 },
+        { "label": "k3b", "matrix": [3, 11], "w": 1, "x": 11, "y": 3 }
+      ]
+    },
+    "LAYOUT_planck_1x2uC": {
+      "c_macro": true,
+      "filename": "keyboards/planck/thk/thk.h",
+      "key_count": 47,
```
Comment (fauxpark) [inline keyboards/planck/thk/info.json:106]:
```suggestion
```

File: keyboards/planck/thk/info.json [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  "community_layouts": ["ortho_4x12", "planck_mit"],
+  "layout_aliases": {
+    "LAYOUT_planck_grid": "LAYOUT_ortho_4x12",
+    "LAYOUT_planck_mit": "LAYOUT_planck_1x2uC"
+  },
+  "layouts": {
+    "LAYOUT_ortho_4x12": {
+      "c_macro": true,
+      "filename": "keyboards/planck/thk/thk.h",
+      "key_count": 48,
```
Comment (fauxpark) [inline keyboards/planck/thk/info.json:51]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/planck/thk/config.h [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+#define DEVICE_VER 0x0000
+#undef PRODUCT
+#define PRODUCT Planck THK
```
Comment (jackhumbert) [inline keyboards/planck/thk/config.h:24]:
```suggestion
#define PRODUCT Planck THK
#define PRODUCT_ID 0x25A7
```

File: keyboards/planck/planck.c [hunk 6]
```diff
@@ -12,8 +12,8 @@ const keypos_t hand_swap_config[MATRIX_ROWS][MATRIX_COLS] = {
 
 __attribute__ ((weak))
 void matrix_init_kb(void) {
-	// Turn status LED on
-	#ifdef __AVR__
+	// Turn status LED on, with the exception of THK
+	#if defined(__AVR__) && !defined(__AVR_ATmega32A__)
```
Comment (drashna) [inline keyboards/planck/planck.c:16]: Might as well just make it for the atmega32u4.  The planck light doesn't use E6 for the LED (it uses D6), and the rest of the planck revisions dont' use it either.   rev1-5 all use atmega32u4's. 
```suggestion
	#if defined(__AVR_ATmega32U4__)
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**